src/app/(main)/(pages)/layout.tsx

```diff
@@ -1,9 +1,16 @@
 import React from 'react';
+import { getServerSession } from 'next-auth';
+import { Redirect } from '@/components/Redirect';
 
 interface Props {
   children: React.ReactNode;
 }
 
-export default function MainLayout(props: Props) {
+export default async function MainLayout(props: Props) {
+  const session = await getServerSession();
```

Comment:
this check will be added at /app/(main)/layout.tsx

It works here as well but you can see the screen for a second and then it redirects, but if you add this check in the layout above, it'll directly redirect without showing the screen for a split second.

Reply:
@ItsFlash10 it is the same. since page is a route group. it will not matter plus my thought was to project the pages only not everything inside the main folder.

Reply:
both the layouts serve the same set of pages and yes it works as I mentioned but the issue is the redirections delay but its fine here as well